### PR TITLE
Make some improvements to Slack mention callbacks

### PR DESCRIPTION
**Pull request by acaire**
* Prevent chomping of colons following mentions
* Use regex to find mentions that are surrounded
  by non-alphanumeric characters
* Fix a bug causing all mentions to be returned
  as a duplicate of the first mention
* Move mention processing to separate function
  and write some tests

### CODE REVIEW

File: errbot/backends/slack.py [hunk 1]
```diff
@@ -793,6 +779,32 @@ def sanitize_uris(text):
...
+        """
+        mentioned = []
+
+        m = re.findall('<@[^>]*>*', text)
+
+        for word in m:
+            try:
+                identifier = self.build_identifier(word)
+            except Exception as e:
+                log.debug("Tried to build an identifier from '%s' but got exception: %s", word, e)
```
Comment (gbin) [inline errbot/backends/slack.py]: Maybe makes that a warn otherwise people might miss it.

Comment (gbin) [inline errbot/backends/slack.py]: Maybe not, the original code was not so I guess it was not a problem.

**Comment by gbin** [comment]
Awesome, thanks !

**Comment by acaire** [comment]
Appreciate it, thanks! :)

**Pull request merged**